BedrockServer.cpp

```diff
@@ -1776,22 +1781,48 @@ void BedrockServer::_control(unique_ptr<BedrockCommand>& command) {
             _maxConflictRetries.store(maxConflictRetries);
         }
     } else if (SIEquals(command->request.methodLine, "BlockWrites")) {
-        shared_ptr<SQLitePool> dbPoolCopy = _dbPool;
-        if (dbPoolCopy) {
-            SQLiteScopedHandle dbScope(*_dbPool, _dbPool->getIndex());
-            SQLite& db = dbScope.db();
-            db.exclusiveLockDB();
+        atomic<bool> locked(false);
+        lock_guard lock(__quiesceLock);
+        if (__quiesceThread) {
+            response.methodLine = "400 Already Blocked";
         } else {
-            response.methodLine = "404 DB Pool Not Found";
+            __quiesceThread = new thread([&]() {
+                shared_ptr<SQLitePool> dbPoolCopy = _dbPool;
+                if (dbPoolCopy) {
+                    SQLiteScopedHandle dbScope(*_dbPool, _dbPool->getIndex());
+                    SQLite& db = dbScope.db();
+                    db.exclusiveLockDB();
+                    locked = true;
+                    while (true) {
+                        if (__quiesceShouldUnlock) {
+                            db.exclusiveUnlockDB();
+                            __quiesceShouldUnlock = false;
+                            return;
+                        }
+
+                        // Wait 10ms for the next check.
+                        usleep(10'000);
+                    }
+                }
+            });
+
+            // Repeatedly wait 10ms for the lock until the thread indicates it's been acquired.
+            while (locked == false) {
+                usleep(10'000);
+            }
```

Comment:
Should we break this infinite loop after 10 seconds or something? If it can't lock in that time, something bigger is probably wonky.

Reply:
It doesn't matter, it's only going to run once 99.99% of the time,the wait is just because we don't know exactly how long the other thread takes to start.